src/public/modules/forms/admin/controllers/admin-form.client.controller.js

```diff
@@ -390,4 +398,72 @@ function AdminFormController(
       return field._id === fieldId
     })
   }
+
+  /**
+   * Checks if a logic unit has any errors in its field values.
+   * @param {Object} logicUnit
+   */
+  const hasInvalidValues = (logicUnit) => {
+    return logicUnit.conditions.some((condition) => {
+      return $scope.checkIfHasInvalidValues(
+        condition.value,
+        $scope.myform.form_fields.find(
+          (field) => field._id === condition.field,
+        ), // must use the myform.form_fields and not condition.fieldInfo because condition.fieldInfo has not been updated
+        condition.state,
+      )
+    })
+  }
+
+  $scope.checkIfHasInvalidValues = function (values, field, state) {
+    if (
```

Comment:
can we extract this functionality in the logic module so that we can reuse it in React?